src/freertos_drivers/esp32/Esp32WiFiManager.cxx

```diff
@@ -1287,15 +1307,8 @@ StateFlowBase::Action Esp32WiFiManager::WiFiStackFlow::init_wifi()
     // or mDNS systems.
     parent_->wifiStatusEventGroup_ = xEventGroupCreate();
 
-    wifi_mode_t requested_wifi_mode = parent_->wifiMode_;
-    if (parent_->wifiMode_ == WIFI_MODE_AP)
-    {
-      // override the wifi mode from AP only to AP+STA so we can perform wifi
-      // scans on demand.
-      requested_wifi_mode = WIFI_MODE_APSTA;
-    }
     // Set the requested WiFi mode.
-    ESP_ERROR_CHECK(esp_wifi_set_mode(requested_wifi_mode));
+    ESP_ERROR_CHECK(esp_wifi_set_mode(parent_->wifiMode_));
```

Comment:
Have you tested that when in AP mode that WiFi scan works? I had tested it long ago and found that it is required to use APSTA mode for scan to work this the comment in the section that was removed above.